packages/starksheet-webapp/src/provider/EVMProvider.ts

```diff
@@ -0,0 +1,89 @@
+import { ChainProvider } from "./chainProvider";
+import { ApplicationBinaryInterface, ContractCall, TransactionReceipt } from "../types";
+import { Web3Provider } from "@ethersproject/providers";
+import { Contract } from "ethers";
+
+/**
+ * Represents an EVM-compatible implementation of the chain provider.
+ */
+export class EVMProvider implements ChainProvider {
```

Comment:
A naïve implementation for EVM. I was not able to test this extensively because the code doesn't run properly yet (see above, ties to starknet remaining in other parts of the code). This should however provide a good basis and not require major tweaks when the code runs properly